### DISCUSSION TITLE
Winforms Controls total Redesign

### DISCUSSION THREAD

**Discussion by hassaanmasood** [dotnet/winforms#6709] [Ideas] [answered]
**Is your feature request related to a problem? Please describe.**

Winforms controls look and feel does not align with Windows 11 and they look really really bad than ever before.
 
**Describe the solution you'd like and alternatives you've considered**

Since the introduction of WPF then UWP and now WinUI, we understand that WinForms is not on the priority list and thus is not updated regularly. But by now Microsoft should realize that many third-party app companies/devs are not willing to make a shift and a lot of apps still using WinForms Style Controls. Windows itself still has many screens and dialogs which is using old UI. 

I believe now is the time Microsoft and Dev Team at WinForms should start investing in a complete overhaul of Winforms Controls and align them with WinUI/Windows 11 look and feel as much as possible and also implement Dark and Light Theme for all controls. Resolving this has a lot of potential and a lot of apps will automatically start aligning to Windows 11 look and feel and it will be much much easier for Devs too.

Sorry but Microsoft Windows has long reigned and only taken advantage of the availability of much more apps on their platform compared to Mac and Linux. But it is very clear that MacOS and Linux provide much more consistent experiences in terms of UI and user experience. 

I understand that Microsoft and .Net Team has put a lot of effort and resources into WPF and UWP also but a better future plan would be:

> Winforms Controls Overhaul for now and provide support till 2027 only.
> Merge the Best of WinForms, WPF and UWP into WinUI. Use WinUI as long term UI framework and only add new features and improvements to it rather than introducing something new every few years.
> Discard WPF and UWP in favour of WinUI. 

Please consider something similar like this. There should be proper guidelines for UX/UI design and Apps not following them should be restricted Windows has become like Andriod where everyone is implementing their own UI completely forgetting about a good user experience. 

The goal for Microsoft should be to make UX/UI consistent across Windows and Apps so Microsoft can retain and only increase its customer base in the long term.

**Will this feature affect UI controls?**
Yes

**Comment by merriemcgaw** [comment]
I'm going to convert this to a discussion where we can talk about options and suggestions.

**Comment by BuslikDrev** [comment]
[Rus]
Лично я хочу, чтобы была технология на подобии [maui](https://github.com/dotnet/maui/discussions/2576), только с поддержкой windows xp+ или возможность в с# указывать условие, какую технологию я применяю для данной системы, чтобы компилятор не ругался на то, что нет такого пространства имён, не может преобразовать тип и т.д.
[Eng]
Personally, I want a technology like [maui](https://github.com/dotnet/maui/discussions/2576), only with windows xp+ support, or the ability to specify in c# the condition which technology I use for this system , so that the compiler does not swear that there is no such namespace, cannot convert the type, etc.

```
if (WINXP) {
System.Windows.Forms.Form _form = new System.Windows.Forms.Form();
Namespace.MyClass.Form = _form;
Namespace.MyClass(_form);
Android.App.LoadApplication(_form);
} else if (ANDROID) {
Xamarin.Forms.Application _form = new Xamarin.Forms.Application();
Namespace.MyClass.Form = _form;
Namespace.MyClass(_form);
Xamarin.Forms.LoadApplication(_form);
} else {
System.Windows.Application _form = new System.Windows.Application();
Namespace.MyClass.Form = _form;
Namespace.MyClass(_form);
System.Windows.Application.Run(_form);
}
```

**Comment by KlausLoeffelmann** [comment, marked as answer]
Please note, that WinForms is in very many areas a wrapper around Windows Win32 controls. If those change, WinForm's look and feel changes. If they remain, WinForm's look and feel remains. We have limited control over the actual look and feel. Please also note that many UI Controls which you see in Windows these days are not based on those Win32 Windows controls, but on Windows WinUI implementation, which we're currently not wrapping in Windows Forms.

[We still have plans to address a series of look-and-feel issues](https://github.com/dotnet/winforms/issues/7641) but are depending on some APIs in Windows and on accessibility compatibility, all of which needs to be carefully evaluated - so there is always literally more to it than the eye can see. So, while we're have that on our radar, we can't make any promises about the scope and a potential release date. 

Upvoting the issue certainly helps so we know the demand for it!
